### PR TITLE
br: add a test for cached table

### LINKED ISSUE #31190: ignore the TableCacheStatus table info in `br`, so it's handled as a normal table

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #31190

Problem Summary:

Handle the compatibility for the cached table feature with `br`

### What is changed and how it works?

`TableCacheStatus` attribute of table info is ignored, so a cached table is handled as a normal table.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by tiancaiamao** [comment]
A subtask for https://github.com/pingcap/tidb/issues/25293

File: br/tests/br_cache_table/run.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+DB="$TEST_NAME"
+
+trim_sql_result() {
```
Comment (3pointer) [inline br/tests/br_cache_table/run.sh:20]: this function can be removed.

Comment (tiancaiamao) [reply br/tests/br_cache_table/run.sh]: Done

File: br/tests/br_cache_table/run.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+run_sql "drop schema $DB;"
+
+echo "restore start..."
+run_br restore db --db $DB -s "local://$TEST_DIR/$DB" --pd $PD_ADDR
+
+set -x
+
+run_sql "select count(*) from $DB.cache_1;"
+check_contains 'count(*): 3'
+
```
Comment (lcwangchao) [inline br/tests/br_cache_table/run.sh:38]: add a run_sql "alter table $DB.cache_1 cache" to ensure the imported table is not cached ?

Comment (tiancaiamao) [reply br/tests/br_cache_table/run.sh:38]: OK, I check the `information_schema.tables` not to make sure the new table is not a cached table.

**Review by lcwangchao — APPROVED** [review]

**Comment by 3pointer** [comment]
```
[2022-03-22T10:10:50.943Z] + run_sql 'select create_options from information_schema.tables where table_schema = br_cache_table and table_name = '\''cache_1'\'';'

[2022-03-22T10:10:50.943Z] ERROR 1054 (42S22) at line 1: Unknown column 'br_cache_table' in 'where clause'
```

**Review by xhebox — APPROVED** [review]

**Pull request merged**